cmake/app/boilerplate.cmake

```diff
@@ -274,6 +274,36 @@ set(KERNEL_EXE_NAME   ${KERNEL_NAME}.exe)
 set(KERNEL_STAT_NAME  ${KERNEL_NAME}.stat)
 set(KERNEL_STRIP_NAME ${KERNEL_NAME}.strip)
 
+# Try to populate USER_HOME_DIR with a directory that user
+# applications may write to. Leave it unset if unable.
+foreach(env_var
+    HOME
+    LOCALAPPDATA
+    )
+  if(DEFINED ENV{${env_var}})
+    set(dir $ENV{${env_var}})
+
+    check_if_directory_is_writeable(${dir} ok)
+    if(${ok})
+      # The directory is write-able
+      set_ifndef(USER_HOME_DIR ${dir})
+      break()
+    else()
+      # Writing to the directory failed, keep looking for a suitable
+      # directory
+    endif()
+  endif()
+endforeach()
+
+# Populate USER_CACHE_DIR with a suitable directory for caching
+# files. Prefer a directory outside of the git repository because it
+# is good practice to have clean git repositories.
+if(DEFINED USER_HOME_DIR)
+  set_ifndef(USER_CACHE_DIR ${USER_HOME_DIR}/.cache/zephyr)
+else()
+  set_ifndef(USER_CACHE_DIR ${ZEPHYR_BASE}/.dir/cache)
```

Comment:
Maybe name this `.cache` for consistency with the one in $HOME? `.dir` may be a little vague.

Reply:
Done